riptide-core/src/main/java/org/zalando/riptide/Requester.java

```diff
@@ -109,7 +113,18 @@ public final Requester contentType(final MediaType contentType) {
 
                 // TODO why not return CompletableFuture<ClientHttpResponse> here?
                 // we need a CompletableFuture<Void>
-                return future.thenApply(response -> null);
+
+                // TODO: replace with thenApply call in Java 9
+                final CompletableFuture<Void> result = preserveCancelability(future);
+                future.whenComplete((response, throwable) -> {
```

Comment:
`future.whenComplete(forwardTo(result))`?

Reply:
Won't work, since this is a poor man's `thenApply`, i.e. I'm passing `null` instead of response.

Reply:
True here. But why?

Reply:
Because the API expects a `CompletableFuture<Void>`. I already have the plan to change this in version 3.0, see #290.